### ISSUE TITLE
IRequestStartup not being registered during RegisterAndGetRequestStartupTasks

### ISSUE THREAD

**Issue by Stephen2** [NancyFx/Nancy.Bootstrappers.StructureMap#27]
Hi everyone, please excuse complete ignorance on how to properly use Github/report issues, and I don't feel confident that my code is good enough to do a pull request (if I could figure out how!!)

Anyway, I upgraded from 0.23 to 1.1 and my external Module, that implemented IRequestStartup wasn't being called anymore.

My code change to fix was:
In this RegisterAndGetRequestStartupTasks function, it appeared like the types weren't actually being registered, so I added:

container.Configure(registry => {
	foreach (var requestStartupType in requestStartupTypes) {
		registry.For(typeof(IRequestStartup)).LifecycleIs(Lifecycles.Unique).Use(requestStartupType);
	}
});

Thanks for being patient with the bad way to get issues reported/code fixes in,
Stephen

**Comment by khellang** [comment]
Yeah, looks like @gregsochanik removed that in 904ced8977e2d00e1b4698840aaa3d44eccc8930. Don't know how that's supposed to work. // @asbjornu

**Comment by asbjornu** [comment]
Yea, it seems like we're missing a `Scan()`. It would be awesome if NancyFx/Nancy#1712 could be resolved and a test suite could be created so problems like these could be discovered in a test rather than in someone's production code during runtime.

**Comment by gregsochanik** [comment]
Ah - sorry, yes, I had a feeling this might happen. 

In the original PR (https://github.com/NancyFx/Nancy.Bootstrappers.StructureMap/pull/24) I asked for some confirmation of how it was supposed to work (quote from PR comments):

"*This is the section I'm not sure about and would like some advice on if you've got time.*

*Looking at the wiki and at other DI Bootstrapper implementations (AutoFaq / Castle etc) it would seem this is added to enforce a distinction between "PerRequest" startups and "Application" startups, specifically to enable PerRequest IRequestStartups to be stored alongside a Request specific nested Container.*

*Have I got this right? If so, I may need to revert this change, leaving the container.Configure() code in.*"

Looks like I hadn't got this right :smile:

@asbjornu I can submit a PR with a fix if you like?

**Comment by asbjornu** [comment]
Yes, please do, @gregsochanik.

**Comment by gregsochanik** [comment]
Done: https://github.com/NancyFx/Nancy.Bootstrappers.StructureMap/pull/31

**Comment by khellang** [comment]
Should be closed with #32 :+1:

**Comment by jvandertil** [comment]
I don't think that is the best solution to the problem. Since the concrete types are passed in as a Type[] we can simply ask StructureMap to give us instances.

Something like this:
```
return requestStartupTypes.Select(container.GetInstance).Cast<IRequestStartup>();
```

This avoids having to configure the container on each request.
